scripts/west_commands/sdk.py

```diff
@@ -0,0 +1,528 @@
+                """
+            Listing SDKs:
+
+                Run 'west sdk' or 'west sdk list' to list installed SDKs.
+                See 'west sdk list --help' for details.
+
+
+            Installing SDK:
+
+                Run 'west sdk install' to install Zephyr SDK.
+                See 'west sdk install --help' for details.
+            """
+            ),
+        )
+
+        subparsers_gen = parser.add_subparsers(
+            metavar="<subcommand>",
+            dest="subcommand",
+            help="select a subcommand. If omitted, treat it as the 'list' selected.",
+        )
+
+        subparsers_gen.add_parser(
+            "list",
+            help="list installed Zephyr SDKs",
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            epilog=textwrap.dedent(
+                """
+            Listing SDKs:
+
+                Run 'west sdk' or 'west sdk list' command information about available SDKs is displayed.
+            """
+            ),
+        )
+
+        install_args_parser = subparsers_gen.add_parser(
+            "install",
+            help="install Zephyr SDK",
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            epilog=textwrap.dedent(
+                """
+            Installing SDK:
+
+                Run 'west sdk install' to install Zephyr SDK.
+
+                Set --version option to install a specific version of the SDK.
+                If not specified, the install version is detected from "${ZEPHYR_BASE}/SDK_VERSION file.
+                SDKs older than 0.14.1 are not supported.
+
+                You can specify the installation directory with --install-dir.
+                If the specified version of the SDK is already installed,
+                the already installed SDK will be used regardless of the settings of
+                --install-dir.
+
+                Typically, Zephyr SDK archives contain only one directory named zephyr-sdk-<version>
+                at the top level.
+                The SDK archive is extracted to the home directory if --install-dir is not specified.
+                In this case, SDK will install into ${HOME}/zephyr-sdk-<version>.
+                If --install-dir is specified, the `zephyr-sdk-<version>` will be renamed and relocated to
+                the specified location.
+
+                --interactive, --toolchains, --no-toolchains and --no-hosttools options
+                specify the behavior of the installer. Please see the description of each option.
+
+                --personal-access-token specifies the GitHub personal access token.
+                This helps to relax the limits on the number of REST API calls.
+
+                --api-url specifies the REST API endpoint for GitHub releases information
+                when installing the SDK from a different GitHub repository.
+            """
+            ),
+        )
+
+        install_args_parser.add_argument(
+            "-v",
```

Comment:
please remove the short-arg `-v` because `west` itself has already established the meaning of `-v` as:
```
$ west -h
usage: west [-h] [-z ZEPHYR_BASE] [-v] [-V] <command> ...

The Zephyr RTOS meta-tool.

optional arguments:
...
  -v, --verbose         Display verbose output. May be given multiple times
                        to increase verbosity.
```
so although the west `-v` and the `sdk` command argument `-v` must be placed at different locations, then it's still slightly confusing that the `-v` can mean different things depending on placement, for example:
```
$ west -v sdk install -v 0.16.0
        ^              ^-- means version
        |----------------- means verbose
```
and yes, the above is indeed a valid command to run :slightly_smiling_face:

Reply:
I don't think this is a bad specification, but since specifying the version is a case where there is a clear intention, I think it would be better to explicitly specify it with the --version. I have removed the -v option.